### ISSUE TITLE
Mapping doesn't await child using AfterMappingAsync (Async package) 

### ISSUE THREAD

**Issue by Bjorn-Anamatch** [MapsterMapper/Mapster#416]
When mapping objects using the Async package:

```
        await _mapper.From(result).AdaptToTypeAsync<ParentDto>();
```

We see that the mapping is not complete when the result is already returned.

```
        config.NewConfig<Parent, ParentDto>()
            .Map(dest => dest.ParentDetailsInternal, src => src.ParentDetails)
            .AfterMappingAsync(AfterMappingParentToDto)
            ;

        config.NewConfig<ParentDetail, ParentDetailDto>()
            .Include<ParentDetailSubType, ParentDetailSubTypeDto>()
            .AfterMappingAsync(AfterMappingChildToDto)
            ;
```

It seems the .Map method is not expecting _async_ methods further down the mapping chain. The AfterMappingAsync on the child is not yet finished.

**Comment by andrerav** [comment]
Good catch there, looking forward to the PR. I will ensure it gets merged quickly and push a prerelease package for you.

**Comment by gdoron** [comment]
Hi, I got to this thread while try to figure out if we should migrate from automapper to mapster.
Basically the biggest motivation for us it async mappers.
Does \ will it exist in mapster?

Thanks!

**Comment by stagep** [comment]
In order to identify issues that are still active, we are closing issues that we believe are either resolved or are dormant. If your issue is still active then please reopen. Thanks.
